tests/realtikvtest/txntest/txn_state_test.go

```diff
@@ -167,9 +167,16 @@ func TestRunning(t *testing.T) {
 		tk.MustExec("commit;")
 		ch <- struct{}{}
 	}()
-	time.Sleep(100 * time.Millisecond)
-	info := tk.Session().TxnInfo()
-	require.Equal(t, txninfo.TxnRunning, info.State)
+	var state txninfo.TxnRunningState
+	for i := 0; i < 1000; i++ {
+		time.Sleep(time.Microsecond)
+		info := tk.Session().TxnInfo()
+		if info != nil && info.State == txninfo.TxnRunning {
+			state = txninfo.TxnRunning
+			break
+		}
+	}
```

Comment:
Besides, could we extract these code into a function? There are three places where you can use this function.